modules/concepts/commands.md

```diff
@@ -77,6 +77,66 @@ services:
 
 The command should be now available using `./bin/console your-module:export`.
 
+
+### Autoloading
+
+The instructions above are assuming the module is installed via composer. In case you follow the same procedure but wish to add the command in your already existing module in `modules/<module_directory>` then the procedure requires an extra step. 
+
+For example with a structure like this 
+
+```tree
+modules/module_directory/
+├── config
+│   └── services.yml
+├── config.xml
+├── module_directory.php
+├── src
+│   └── Command
+│       └── SyncCommand.php
+```
+
+You would also need to namespace your src directory in `docroot/composer.json`
+
+By appending the following 
+
+```
+    "autoload": {
```

Comment:
Are you talking about the root composer.json file ? https://github.com/PrestaShop/PrestaShop/blob/develop/composer.json

Reply:
Yes

Reply:
Also I don't think it's actually a bug, it's more like a symfony command is declared outside of the prestashop core bundle so it's normal that the namespace is unknown to external modules unless you declare it.

I guess creating a composer plugin that extends the composer autoloader to find those namespaces when composer is dumping the autoloader would automate this but I don't want to refer to missing functionality as a bug

Reply:
Thank you for explanation.
You are right this is a missing functionality rather than a bug, but nonetheless this is something PrestaShop should be able to do one day.

As this piece of doc you have written is a nice workaround the current limitation I'm not sure we should put it in the Command page. Maybe in the FAQ section it would be better? https://devdocs.prestashop.com/1.7/faq/

What you have written has value 👍  but it's a how-to that aim to help others looking to do it. It's away you could find by experience to make things work but the standard doc should talk about 'standard and recommended ways' I guess

wdyt @PrestaShop/prestashop-maintainers

Reply:
Thank you, @ioweb-gr - this can be useful for some.

In my opinion, it's more like a workaround than a preferred way of doing things, and I think it would be better to have it in the FAQ section. I'll leave the final decision to @thomasnares

Reply:
Yes it should be definitely in FAQ section, can you move it to FAQ @ioweb-gr ?
Thanks 👍